CMakeLists.txt

```diff
@@ -67,6 +67,9 @@ ADD_CLANG_TIDY()
 #--- Declare options -----------------------------------------------------------
 option(CREATE_DOC "Whether or not to create doxygen doc target." OFF)
 option(ENABLE_SIO "Build SIO I/O support" OFF)
+if (ENABLE_SIO)
+  add_compile_definitions(PODIO_WITH_SIO)
+endif()
```

Comment:
Can we move this to the `tests/CMakeLists.txt` or even better as a simple `target_compile_definitions` to the unittests? Otherwise we "pollute" all of our targets with this definition, and I would like to avoid that (at least for the moment).